src/checkers/inference/InferenceMain.java

```diff
@@ -339,8 +358,17 @@ public BaseAnnotatedTypeFactory getRealTypeFactory() {
 
     public SlotManager getSlotManager() {
         if (slotManager == null ) {
-            slotManager = new DefaultSlotManager(inferenceChecker.getProcessingEnvironment(),
-                    realTypeFactory.getSupportedTypeQualifiers(), true );
+            Set<? extends AnnotationMirror> tops = realTypeFactory.getQualifierHierarchy().getTopAnnotations();
+            if (tops.size() != 1) {
```

Comment:
In PICO I think we have two tops, one for immutability and one for initialization. Only one hierarchy, immutability, is inferred, the other one is not. Haifeng will run in problems with this, but I'm fine with solving it then.

Reply:
Ok, we should discuss more about this in his PR.